decidim-core/app/controllers/decidim/notifications_subscriptions_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Decidim
+  # The controller to handle the subscriptions to push notifications
+  class NotificationsSubscriptionsController < Decidim::ApplicationController
+    skip_before_action :verify_authenticity_token
+
+    def create
+      return unless current_user
+
+      Decidim::NotificationsSubscription.find_or_create_by(auth: params[:keys][:auth]) do |subscription|
+        subscription.decidim_user_id = current_user.id
+        subscription.endpoint = params[:endpoint]
+        subscription.p256dh = params[:keys][:p256dh]
+      end
+
+      head :ok
+    end
+
+    def delete_by_user
```

Comment:
Yep. you're right. I changed it. Now it doesn't need any user_id param, it uses current_user. But what do you with with a before_action? what should it do?

Reply:
it's in the pr

Reply:
This seems to be already fixed but the delete action will still throw an exception in case the user is not authenticated.

One way to solve this would be to move the routes under the `authenticate(:user) do ...` block in the routes when these routes would not be available when the user is not authenticated.